// Copyright (c) 2020-2021 The Open-Transactions developers
// This Source Code Form is subject to the terms of the Mozilla Public
// License, v. 2.0. If a copy of the MPL was not distributed with this
// file, You can obtain one at http://mozilla.org/MPL/2.0/.

syntax = "proto2";

package opentxs.proto;
option java_package = "org.opentransactions.proto";
option java_outer_classname = "OTCredential";
option optimize_for = LITE_RUNTIME;

import public "Enums.proto";
import public "ChildCredentialParameters.proto";
import public "MasterCredentialParameters.proto";
import public "KeyCredential.proto";
import public "ContactData.proto";
import public "Signature.proto";
import public "VerificationSet.proto";

message Credential {
    optional uint32 version = 1;
    optional string id = 2;
    optional CredentialType type = 3;
    optional CredentialRole role = 4;
    optional KeyMode mode = 5;
    optional string nymID = 6;
    optional ChildCredentialParameters childData = 7;
    optional MasterCredentialParameters masterData = 8;
    oneof publicData {
        KeyCredential publicCredential = 9;
        ContactData contactData = 11;
        VerificationSet verification = 12;
    }
    oneof privateData {
        KeyCredential privateCredential = 10;
    }
    repeated Signature signature = 15;
}
